### PR TITLE
+(*)MOM_hor_visc bug fix and loop size cleanup

### PR DESCRIPTION
**Pull request by Hallberg-NOAA**
Fixed the bug noted in issue #72 and excessive loop sizes noted in the
unresolved comments in a recent commit, and cleaned up other aspects of
MOM_hor_visc, mostly related to the code that is exercised when USE_GME=True.

 - Fixed the bug with the wrong arrays being used when ADD_LES_VISCOSITY=True
   that was noted in MOM6 issue #72.

 - Corrected some of the overly large loop extents that were noted in unresolved
   comments about MOM6 PR #65.

 - Only log USE_KH_BG_2D if a Laplacian viscosity is used.

 - Use extra calculations in the halos and marching in to avoid unnecessary halo
   updates in smooth_GME if there multiple smoothing passes.

 - Corrected the capitalization of some horizontal indices to follow the MOM6
   convention for indicating the horizontal staggering position.

 - Collected the post_data calls for diagnostics with use_GME with the other
   post data calls to collocate some of the potential inter-PE synchronization
   points.

 - Fixed the indentation of some expressions that were using less than 4 points
   for some continuation lines.

 - Eliminated several unused variables, and fused some loops to allow 2-d
   variables to be replaced with scalars.

  With this PR, answers can change when ADD_LES_VISCOSITY=True and there is a
Smagorinsky or Leith Laplacian viscosity and there is a nonzero background.
One run-time parameter is no longer logged if LAPLACIAN=false, so there are
changes to the MOM_parameter_doc files.  All answers in the MOM6-examples test
suite are bitwise identical.

### CODE REVIEW

**Comment by herrwang0** [comment]
Thanks for addressing the issue regarding `add_LES_viscosity`.

While we are on this, I think there is another issue (https://github.com/mom-ocean/MOM6/issues/1544) with modified_leith  (I posted in the main repo during the migration and I didn't know the right place to post). It looks like the code is slightly different from the description in the comments (and the literatures I think).

File: src/parameterizations/lateral/MOM_hor_visc.F90 [hunk 1]
```diff
@@ -531,20 +501,34 @@ subroutine horizontal_viscosity(u, v, h, diffu, diffv, MEKE, VarMix, G, GV, US,
...
+            (0.25*((dudx_bt(i,j)+dudx_bt(i+1,j+1)) + (dudx_bt(i,j+1)+dudx_bt(i+1,j))))**2 + &
+            (0.25*((dvdy_bt(i,j)+dvdy_bt(i+1,j+1)) + (dvdy_bt(i,j+1)+dvdy_bt(i+1,j))))**2)
+      ! Probably the following test could be simplified to
+      ! if (boundary_mask_q * G%mask2dBu(I,J) > 0.0) then
+      if (grad_vel_mag_bt_q > 0.0) then
         h_arith_q = 0.25 * ((htot(i,j) + htot(i+1,j+1)) + (htot(i+1,j) + htot(i,j+1)))
         I_hq = 1.0 / h_arith_q
         h_harm_q = 0.25 * h_arith_q * ((htot(i,j)*I_hq + htot(i+1,j+1)*I_hq) + &
                                        (htot(i+1,j)*I_hq + htot(i,j+1)*I_hq))
+        !### The two expressions above looks like they simplify to just the arithmetic mean!
```
Comment (gustavo-marques) [inline src/parameterizations/lateral/MOM_hor_visc.F90:530]: Good catch! Indeed, these expressions simplify to the arithmetic mean in line 528. However, answers changed when I used `h_harm_q = 0.25 * ((htot(i,j) + htot(i+1,j+1)) + (htot(i+1,j) + htot(i,j+1)))`. Any idea why? Regardless, we do not need to maintain answers for USE_GME = True` at this time, so I think we should we the simplified expression.

Comment (Hallberg-NOAA) [reply src/parameterizations/lateral/MOM_hor_visc.F90:530]: The reason why the answers change is simple.  At roundoff, the expression a * (1/a) is not exactly one.  We have seen this many times where replacing division by the multiplication by the reciprocal of the denominator changes answers at roundoff.  However, the guidance that we can simplify this expression now (before it is more widely used) and can tolerate answer changes is perfect.  I will go ahead and make the simplification.

**Comment by Hallberg-NOAA** [comment]
I believe that the latest commit should satisfactorily address the suggestion by @gustavo-marques, and that this PR is now ready for pipeline testing.

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/15005 :heavy_check_mark: :yellow_circle: 

Removed parameter:
* `USE_KH_BG_2D` (now conditionally logged)

**Comment by marshallward** [comment]
I can't see any reason what status GitHub is waiting on here, and I've tried the "resubmit" trick to no avail, so I am going to admin-override this PR.

**Pull request merged**